patches/server/0950-Avoid-respawn-desync-of-player-metadata.patch

```diff
@@ -0,0 +1,86 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lulu13022002 <41980282+Lulu13022002@users.noreply.github.com>
+Date: Sat, 17 Dec 2022 13:04:14 +0100
+Subject: [PATCH] Avoid respawn desync of player metadata
+
+
+diff --git a/src/main/java/net/minecraft/network/syncher/SynchedEntityData.java b/src/main/java/net/minecraft/network/syncher/SynchedEntityData.java
+index 37e193f57938047c8b886ed7d2816411392f94b4..f7d1425d102959f7a2f1da5585e2d3d2032a5abb 100644
+--- a/src/main/java/net/minecraft/network/syncher/SynchedEntityData.java
++++ b/src/main/java/net/minecraft/network/syncher/SynchedEntityData.java
+@@ -152,6 +152,15 @@ public class SynchedEntityData {
+         this.isDirty = true;
+     }
+     // CraftBukkit end
++    // Paper start
++    public <T> void markDirtyIfNotDefault(EntityDataAccessor<T> datawatcherobject) {
++        DataItem<T> item = this.getItem(datawatcherobject);
++        if (!item.isSetToDefault()) {
++            item.setDirty(true);
++            this.isDirty = true;
++        }
++    }
++    // Paper end
+ 
+     public boolean isDirty() {
+         return this.isDirty;
+diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+index c0b0a7fdb75266a7064d54bda6441953184ecc64..96f4be56db7d02e4d6c342b5c0dbea7958f9f564 100644
+--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
++++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+@@ -1776,31 +1776,35 @@ public class ServerPlayer extends Player {
+         if (alive) {
+             this.getInventory().replaceWith(oldPlayer.getInventory());
+             this.setHealth(oldPlayer.getHealth());
++            this.getEntityData().markDirtyIfNotDefault(LivingEntity.DATA_HEALTH_ID); // Paper
++            this.getEntityData().markDirtyIfNotDefault(Player.DATA_PLAYER_ABSORPTION_ID); // Paper
+             this.foodData = oldPlayer.foodData;
+             this.experienceLevel = oldPlayer.experienceLevel;
+             this.totalExperience = oldPlayer.totalExperience;
+             this.experienceProgress = oldPlayer.experienceProgress;
+-            this.setScore(oldPlayer.getScore());
++            this.getEntityData().markDirtyIfNotDefault(Player.DATA_SCORE_ID); // Paper
```

Comment:
I think the old setXXX(getXXX()) stuff should stay, just after it you mark it as dirty if its not default. In case it ever changes upstream and we don't catch it here